fdbserver/RESTKmsConnector.actor.cpp

```diff
@@ -1473,6 +1508,38 @@ void testGetBlobMetadataRequestBody(Reference<RESTKmsConnectorCtx> ctx) {
 	}
 }
 
+void testMissingOrInvalidVersion(Reference<RESTKmsConnectorCtx> ctx, bool isCipher) {
+	rapidjson::Document doc;
+	doc.SetObject();
+
+	rapidjson::Value cDetails(rapidjson::kArrayType);
+	rapidjson::Value detail(rapidjson::kObjectType);
+	rapidjson::Value key(isCipher ? BASE_CIPHER_ID_TAG : BLOB_METADATA_DOMAIN_ID_TAG, doc.GetAllocator());
+	rapidjson::Value id;
+	id.SetUint(12345);
+	detail.AddMember(key, id, doc.GetAllocator());
+	cDetails.PushBack(detail, doc.GetAllocator());
+	key.SetString(isCipher ? CIPHER_KEY_DETAILS_TAG : BLOB_METADATA_DETAILS_TAG, doc.GetAllocator());
+	doc.AddMember(key, cDetails, doc.GetAllocator());
+
+	rapidjson::Value versionKey(REQUEST_VERSION_TAG, doc.GetAllocator());
+	rapidjson::Value versionValue;
+	int version = INVALID_REQUEST_VERSION;
+	if (deterministicRandom()->coinflip()) {
+		if (deterministicRandom()->coinflip()) {
+			version = -7;
+		} else {
+			version = (isCipher ? SERVER_KNOBS->REST_KMS_CURRENT_CIPHER_REQUEST_VERSION
+			                    : SERVER_KNOBS->REST_KMS_CURRENT_BLOB_METADATA_REQUEST_VERSION) +
+			          10;
+		}
+	} else {
+		// set to invalid_version
+	}
+	versionValue.SetInt(version);
```

Comment:
This test isn't asserting anything. should we call `parseEncryptCipherResponse`/`parseBlobMetadataResponse`?